src/routes/api/users.js

```diff
@@ -1,21 +1,88 @@
+import Middlewares from '../../middlewares';
+import UserController from '../../controllers/UserController';
+
+const { forgotPasswordRules, resetPasswordRules } = Middlewares;
 const router = require('express').Router();
 
+const {
+  checkForgotPasswordMiddleware,
+  checkResetPasswordMiddleware,
+  decodeTokenMiddleware
+} = Middlewares;
+
+const { forgotPasswordController, resetPasswordController } = UserController;
+router.post('/forgot-password', forgotPasswordRules(), checkForgotPasswordMiddleware, forgotPasswordController);
+router.patch('/reset-password/:token', decodeTokenMiddleware, resetPasswordRules(), checkResetPasswordMiddleware, resetPasswordController);
+
 /**
  * @swagger
- *
- * /users/login:
+ * /users/forgot-password:
  *    post:
- *      summary: User login
+ *      summary: User receives an email with a reset password link once they hit forgot password
  *      tags: [Users]
  *      requestBody:
  *        required: true
  *        content:
  *          application/json:
  *            schema:
- *              $ref: '#/components/schemas/User'
+ *              $ref: '#/components/schemas/ForgotPassword'
  *      responses:
  *        "200":
- *          description: A user schema
+ *          description: 'A user schema'
+ *        "400":
+ *          description: 'Bad request'
+ *        "500":
+ *          description: 'Server Error'
+ * components:
+ *   schemas:
+ *     ForgotPassword:
+ *       type: "object"
+ *       properties:
+ *         email:
+ *           type: string
+ *       required:
+ *         - 'email'
+ *
  */
 
```

Comment:
You can change the name of this file to authRoute.js as @NiyoEric  and @william000000

Reply:
@hezronkimutai I think I should either leave it as users because Eric and William are dealing with user authentication but mine is a feature to do with users or what do you think?

Reply:
@hezronkimutai Actually let me add it. It actually falls under the authentication skill on second thought

Reply:
Okey Ivy